content/javascript/concepts/optional-chaining/optional-chaining.md

```diff
@@ -0,0 +1,167 @@
+---
+Title: 'Optional Chaining'
+Description: 'The optional chanining operator simplifies the comparison of multiple data properties in a chain of connected objects' 
+Subjects: 
+  - 'Web Development'
+  - 'Computer Science'
+Tags: 
+  - 'Data'
+  - 'Operators'
+CatalogContent: 
+  - 'introduction-to-javascript'
+  - 'paths/front-end-engineer-career-path'
+---
+
+
+
+The **optional chanining** operator simplifies the comparison of multiple data properties in a chain of connected objects. This is especially valuable if any of the properties are null as the operatior will return as undefined instead of throwing an error.
+
+Optional chaning can be used to validate nested object properties, call optional functions or methods on optional objects, accessing arrary elements, and more. 
+
+The primary benefits of using optional chaining instead of regular syntax include simplifying the code to achieve the same result so it's easier to read, protecting against runtime errors and enhances maintainability.
+
+## Syntax
+
+The basic syntax for `optional chaining` can be defined as: 
+
+```js
+// To access an object property
+object?.property
+
+//To access an element in an array
+array?.[index]
+
+//To invoke a function (if it exists)
+object?.method?.()
+```
+
+## Example
+This person object lists an individual's details and containing properties, nested objects, an array, and a method. Optional chaining for each conditional evaluation is listed below. 
+
+```js
+const person = {
+  name: "Alice",
+  gender: "Female",
+  age: 12,
+  address: {
+    street: "1111 Palm Ave", 
+    city: "Broken Arrow",
+    state: "Oklahoma"
+  },
+  favoriteFoods: ["pizza", "ice cream", "cake"], 
+  commonPhrase: function() {
+    return `${this.name} always says "I love ${this.address.state}."`;
+  }
+};
+```
+
+## Accessing Object Properties
+To search for the `state` object of `person`:
+
+```js
+//Regular syntax
+const state = person && person.address && person.address.state;
+console.log(`State is: ${state}`);
+
+//This can be rewritten as:
+const state = person?.address?.state;
+console.log(`State is: ${state}`);
+
+//The output for both is: 
+State is: Oklahoma
+```
+
+## Accessing Arrary Elements
+To search for the `cake` in the `favoriteFoods` array of `person`:
+
+```js
+//Regular Syntax
+const food = person && person.favoriteFoods && person.favoriteFoods.find(item => item === "cake");
+console.log(`Favorite Food is: ${food}`);
+
+//This can be rewritten as:
+const food = person?.favoriteFoods.find(item => item === "cake");
+console.log(`Favorite Food is: ${food}`)
```

Comment:
Add complete code, including the definition for the `person` object, so that the learner can quickly copy and paste this into a code editor and run this code like this -


```
const person = {
  name: "Alice",
  gender: "Female",
  age: 12,
  address: {
    street: "1111 Palm Ave", 
    city: "Broken Arrow",
    state1: "Oklahoma"
  },
  favoriteFoods: ["pizza", "ice cream", "cake"], 
  commonPhrase: function() {
    return `${this.name} always says "I love ${this.address.state1}."`;
  }
};

// Regular Syntax for searching favorite food
const foodRegular = person && person.favoriteFoods && person.favoriteFoods.find(item => item === "cake");
console.log(`Favorite Food (regular syntax) is: ${foodRegular}`);

// This can be rewritten as:
const foodChaining = person?.favoriteFoods.find(item => item === "cake");
console.log(`Favorite Food (optional chaining) is: ${foodChaining}`);

```

Output:

```
Favorite Food (regular syntax) is: cake
Favorite Food (optional chaining) is: cake
```